pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserService.java

```diff
@@ -0,0 +1,129 @@
+        DBUtils.checkDelete(instanceIds.size(), instanceDao.deleteAllByIds(instanceIds));
+        // Removing counters, events
+        handle.attach(JdbiEventConfigurationOccurrenceCounter.class).deleteAllByParticipantId(userId);
+        JdbiQueuedEvent queueEvent = handle.attach(JdbiQueuedEvent.class);
+        queueEvent.deleteQueuedNotificationSubstitutionsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedNotificationsByUserIds(Collections.singleton(userId));
+        queueEvent.deleteQueuedEventsByUserIds(Collections.singleton(userId));
+        // Removing governances
+        UserGovernanceDao userGovernanceDao = handle.attach(UserGovernanceDao.class);
+        List<Governance> userGovernances = userGovernanceDao.findGovernancesByParticipantGuid(user.getGuid()).collect(Collectors.toList());
+        Set<String> studyGuids = userGovernances.stream()
+                .map(Governance::getGrantedStudies).flatMap(Collection::stream).map(GrantedStudy::getStudyGuid)
+                .collect(Collectors.toSet());
+        userGovernanceDao.deleteAllGovernancesForProxy(userId);
+        userGovernanceDao.deleteAllGovernancesForParticipant(userId);
+        handle.attach(StudyGovernanceSql.class).deleteAgeUpCandidateByParticipantId(userId);
+        // Removing user itself
+        handle.attach(DataExportDao.class).deleteDataSyncRequestsForUser(userId);
+        handle.attach(JdbiUser.class).deleteAllByIds(Collections.singleton(userId));
+
+        // Cleaning up the elasticsearch data
+        RestHighLevelClient esClient = ElasticsearchServiceUtil.getElasticsearchClient(ConfigManager.getInstance().getConfig());
+        BulkRequest bulkRequest = new BulkRequest().timeout("2m");
+        for (String studyGuid : studyGuids) {
+            StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+
+            String indexParticipant = ElasticsearchServiceUtil.getIndexForStudy(handle, studyDto,
+                    ElasticSearchIndexType.PARTICIPANTS);
+            String indexParticipantStructured = ElasticsearchServiceUtil.getIndexForStudy(handle, studyDto,
+                    ElasticSearchIndexType.PARTICIPANTS_STRUCTURED);
+
+            bulkRequest.add(new DeleteRequest()
+                    .index(indexParticipant)
+                    .type(REQUEST_TYPE)
+                    .id(userGuid));
+
+            bulkRequest.add(new DeleteRequest()
+                    .index(indexParticipantStructured)
+                    .type(REQUEST_TYPE)
+                    .id(userGuid));
+        }
+
+        BulkResponse bulkResponse = esClient.bulk(bulkRequest, RequestOptions.DEFAULT);
+
+        if (bulkResponse.hasFailures()) {
+            LOG.error(bulkResponse.buildFailureMessage());
+        }
+        userGovernances.stream().map(Governance::getProxyUserGuid)
+                .forEach(proxyGuid -> handle.attach(DataExportDao.class).queueDataSync(proxyGuid));
```

Comment:
I'm not sure this will work. I think the issue is that the proxy is not actually "enrolled" in the study, e.g. they don't have a row in the `user_study_enrollment` table, so `DataSyncJob` might skip over them. I think we need to do `queueDataSync(proxyGuid, studyGuid)` in order for the `DataSyncJob` to pick up and process it properly.

Reply:
You're right, I overlooked that.
Fixed.